### ISSUE TITLE
keras.layers.Concatenate could support list inputs with length 1

### ISSUE THREAD

**Issue by hartikainen** [tensorflow/tensorflow#23176]
**System information**
- TensorFlow version (you are using):
```
$ pip freeze | grep tensorflow
tensorflow==1.12.0rc1
tensorflow-estimator==1.10.12
```
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Currently, the `tf.keras.layers.Concatenate` only supports [list inputs with length >= 2](https://github.com/tensorflow/tensorflow/blob/e5c17aef836f8b85591cdcae31fbb66ddcf8185a/tensorflow/python/keras/layers/merge.py#L378).

Considering that the vanilla `tf.concat` works with inputs of length 1, it would be nice if the keras layers implemented the same behavior.

```python
import numpy as np
import tensorflow as tf


def test_tensorflow_concatenate(inputs):
    tf.concat(inputs, axis=-1)

    print("tf.concat works with {} inputs".format(len(inputs)))


def test_concatenate_layer_with_inputs(inputs):
    model = tf.keras.Sequential((
        tf.keras.layers.Concatenate(axis=-1),
        tf.keras.layers.Dense(32)))

    feed_dict = {
        input_: np.random.uniform(
            0, 1, (3, *input_.shape[1:].as_list()))
        for input_ in inputs
    }
    output = model(inputs)
    output_eval = tf.keras.backend.get_session().run(
        output, feed_dict=feed_dict)
    output_np = model.predict([feed_dict[key] for key in inputs])

    assert np.allclose(output_eval, output_np)

    print("tf.keras.layers.Concatenate with {} inputs".format(len(inputs)))


def main():
    input1 = tf.keras.layers.Input((1, ))
    input2 = tf.keras.layers.Input((2, ))

    test_tensorflow_concatenate([input1, input2])
    test_tensorflow_concatenate([input1])

    test_concatenate_layer_with_inputs([input1, input2])
    test_concatenate_layer_with_inputs([input1])


if __name__ == '__main__':
    main()
```

**Will this change the current api? How?**
No ValueErrors would be raised when calling the `tf.keras.layers.Concatenate` with input list of length 1.

**Who will benefit with this feature?**
Anyone who dynamically creates multi-input/-output keras models.

**Any Other info.**
n/a

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
